package.json

```diff
@@ -9,27 +9,27 @@
     "publish-docs": "gh-pages -d docs/jsdocs",
     "start:test": "gulp dev:test",
     "build:test": "gulp build:test",
-    "test": "npm run test:unit && npm run lint",
+    "test": "yarn test:unit && yarn lint",
     "dapp": "static-server test/e2e/contract-test --port 8080",
-    "dapp-chain": "GANACHE_ARGS='-b 2' concurrently -k -n ganache,dapp -p '[{time}][{name}]' 'npm run ganache:start' 'sleep 5 && static-server test/e2e/contract-test --port 8080'",
-    "watch:test:unit": "nodemon --exec \"npm run test:unit\" ./test ./app ./ui",
+    "dapp-chain": "GANACHE_ARGS='-b 2' concurrently -k -n ganache,dapp -p '[{time}][{name}]' 'yarn ganache:start' 'sleep 5 && static-server test/e2e/contract-test --port 8080'",
+    "watch:test:unit": "nodemon --exec \"yarn test:unit\" ./test ./app ./ui",
     "test:unit": "cross-env METAMASK_ENV=test mocha --exit --require test/setup.js --recursive \"test/unit/**/*.js\" \"ui/app/**/*.test.js\"",
     "test:single": "cross-env METAMASK_ENV=test mocha --require test/helper.js",
-    "test:integration": "npm run test:integration:build && npm run test:flat",
+    "test:integration": "yarn test:integration:build && yarn test:flat",
     "test:integration:build": "gulp build:scss",
     "test:e2e:drizzle:beta": "SELENIUM_BROWSER=chrome test/e2e/run-drizzle.sh",
     "test:e2e:chrome": "SELENIUM_BROWSER=chrome test/e2e/run-all.sh",
     "test:web3:chrome": "SELENIUM_BROWSER=chrome test/e2e/run-web3.sh",
     "test:web3:firefox": "SELENIUM_BROWSER=firefox test/e2e/run-web3.sh",
     "test:e2e:firefox": "SELENIUM_BROWSER=firefox test/e2e/run-all.sh",
-    "test:coverage": "nyc --reporter=text --reporter=html npm run test:unit && npm run test:coveralls-upload",
+    "test:coverage": "nyc --reporter=text --reporter=html npm run test:unit && yarn test:coveralls-upload",
```

Comment:
I had to use `npm run` here instead of `yarn` to run the `test:unit` script because of a problem with how `yarn` and `spawn-wrap` (used by `nyc`) are interacting.

`yarn` is creating a shell script proxy for `node` and `yarn` to ensure the version used _within_ these scripts is the same as the version used to _invoke_ the script, which is generally what you want to happen (before this change, it relied upon the `PATH`, which could be messed up in unexpected ways in a variety of circumstances). But `nyc` specifically relies upon wrapping `node` in order to add instrumentation.

I'll link the `yarn` issue when I've created it - this needs a bit more investigation. In the meantime, using `npm run` works just fine.

Reply:
Here is the relevant Yarn issue: https://github.com/yarnpkg/yarn/issues/6746

Reply:
See also: #7994, where I'm updating this